pkg/apis/local/v1alpha1/localvolumeset_types.go

```diff
@@ -0,0 +1,122 @@
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational or nonRotational
+type DeviceMechanicalProperty string
+
+// The mechanical properties of the devices
+const (
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+// DeviceType is the types that will be supported by the LSO.
+type DeviceType string
+
+const (
+	// RawDisk represents a device-type of block disk
+	RawDisk DeviceType = "disk"
+	// Partition represents a device-type of partion
+	Partition DeviceType = "part"
+	// Loop type device
+	Loop DeviceType = "loop"
+)
+
+// DeviceInclusionSpec holds the inclusion filter spec
+type DeviceInclusionSpec struct {
+	// Devices is the list of devices that should be used for automatic detection.
+	// This would be one of the types supported by the local-storage operator. Currently,
+	// the supported types are: disk, part. If the list is empty only `disk` types will be selected
+	// +optional
+	DeviceTypes []DeviceType `json:"deviceTypes,omitempty"`
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperties []DeviceMechanicalProperty `json:"deviceMechanicalProperties,omitempty"`
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize *resource.Quantity `json:"minSize,omitempty"`
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize *resource.Quantity `json:"maxSize,omitempty"`
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models,omitempty"`
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors,omitempty"`
+}
+
+// LocalVolumeSetSpec defines the desired state of LocalVolumeSet
+type LocalVolumeSetSpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// StorageClassName to use for set of matched devices
+	StorageClassName string `json:"storageClassName"`
+	// Maximum number of Devices that needs to be detected per node.
+	// +optional
+	MaxDeviceCount *int32 `json:"maxDeviceCount,omitempty"`
+	// VolumeMode determines whether the PV created is Block or Filesystem. By default it will
+	// be block
+	// +optional
+	VolumeMode localv1.PersistentVolumeMode `json:"volumeMode,omitempty"`
+	// FSType type to create when volumeMode is Filesystem
+	// Mandatory if volumeMode is filesystem.
```

Comment:
I think default is ext4 if unspecified. I do not think there is a way to validate this field one way or the other if `volumeModeis set"

Reply:
fixed